File: Dart/web_scraper_app/lib/objects/itemList.dart
import "item.dart";

class ItemList {
  List<Item> itemList = [];

  ItemList(this.itemList);
  bool addItem(Item item) {
    if (item is Item) {
      this.itemList.add(item);
      return true;
    } else {
      return false;
    }
  }

  bool removeItem(Item item) {
    if (item is Item && this.itemList.contains(item)) {
      this.itemList.remove(item);
      return true;
    } else {
      return false;
    }
  }

  List getItemList() {
    return this.itemList;
  }

  bool checkItemInList(Item item) {
    if (this.itemList.contains(item)) {
      return true;
    } else {
      return false;
    }
  }

  int length() {
    return this.itemList.length;
  }

  Item getItemAt(int index) {
    return this.itemList[index];
  }
}

File: Python/Objects/itemList.py
from item import Item

class Item_List:

    #Need to make list a dict
    def __init__(self):
        self.items_List = []

    def addItem(self,currentItem):
        #Probably needs more checks
        if currentItem is Item:
            self.items_List.append(currentItem)
        else:
            raise Exception("ITEM IS NOT TYPE ITEM")

    def removeItem(self,currentItem):
        if currentItem in self.items_List and currentItem is Item:
            self.items_List.remove(currentItem)
        else:
            raise Exception("ERROR REMOVING ITEM")

    def getItemList(self):
        return self.items_List

    def getItemInfo(self):
        #Todo
        pass
    def findBestPrice(self,currentItem):
        #Need to change to dict first
        pass
